server/services/store/mattermostauthlayer/mattermostauthlayer.go

```diff
@@ -910,7 +910,13 @@ func (s *MattermostAuthLayer) GetBoardsForUserAndTeam(userID, teamID string, inc
 	}
 
 	boards, err := s.Store.GetBoardsInTeamByIds(boardIDs, teamID)
-	if err != nil {
+	// ToDo: check if the query is being used appropriately from the
+	//       interface, Wiggin77 thinks we're passing IDs that belong
+	//       to boards and templates, and the store method only
+	//       fetches boards, so we need to accept a partial result as
+	//       valid
+	var naf *model.ErrNotAllFound
```

Comment:
Why not use the IsNotFound method?  I think there is a IsNotAllFound method as well.

Reply:
I wanted to filter out "full" not founds as in those cases we'd return a `nil` struct rather than an empty one, but we should control those instead of returning the error, I'll fix it

Reply:
Fixed on [146f4c6](https://github.com/mattermost/focalboard/pull/3842/commits/146f4c67b207a29b45a2b91488645fda9105ed23)